In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
hot_hand = pd.read_csv('NBA Hot Hand Analysis(Sheet1) (1).csv')
hot_hand

Players and Stats for 2023-24 Season  Points (per 100)  Rebounds (per 100)  \
0                 Giannis Antetokounmpo              41.6                15.7   
1                          Lebron James              34.6                 9.8   
2                           Bam Adebayo              28.2                15.2   
3                           Luka Doncic              43.3                11.8   
4                          Kevin Durant              35.5                 8.7   
5                     Tyrese Haliburton              29.5                 5.7   
6               Shai Gilgeous-Alexander              42.4                 7.8   
7                        Damian Lillard              33.1                 6.0   
8                          Nikola Jokic              37.8                17.7   
9                          Jayson Tatum              37.1                11.2   
10                        Jalen Brunson              40.9                 5.1   
11                         Devin Booker              36.7                 6.1   
12                         Tyrese Maxey              34.0                 4.8   
13                          Joel Embiid              50.9                16.2   
14                      Anthony Edwards              36.6                 7.7   
15                        Stephen Curry              39.1                 6.6   
16                     Donte Divincezno              26.9                 6.3   
17                         James Harden              23.8                 7.4   
18                        Kawhi Leonard              34.2                 8.8   
19                   Karl-Anthony Towns              32.9                12.6   

    Assists (per 100)  Blocks (per 100)  Steals (per 100)  \
0                 8.9               1.5               1.6   
1                11.2               0.7               1.7   
2                 5.7               1.4               1.7   
3                12.5               0.7               1.8   
4                 6.6               1.6               1.2   
5                16.0               1.0               1.8   
6                 8.8               1.3               2.8   
7                 9.5               0.3               1.3   
8                12.8               1.2               2.0   
9                 6.8               0.8               1.4   
10                9.6               0.2               1.3   
11                9.4               0.6               1.2   
12                8.1               0.6               1.3   
13                8.2               2.5               1.7   
14                7.2               0.7               1.8   
15                7.6               0.6               1.1   
16                4.6               0.8               2.3   
17               12.3               1.1               1.5   
18                5.2               1.3               2.4   
19                4.6               1.0               1.0   

   Field Goal Percentage Three-point Percentage Free Throw Percentage  \
0                 61.10%                 27.40%                65.70%   
1                 54.00%                 41.00%                75.00%   
2                 52.10%                 35.70%                75.50%   
3                 48.70%                 38.20%                78.60%   
4                 52.30%                 41.30%                85.60%   
5                 47.70%                 36.40%                85.50%   
6                 53.50%                 35.30%                87.40%   
7                 42.40%                 35.40%                   92%   
8                 58.30%                 35.90%                81.70%   
9                 47.10%                 37.60%                83.30%   
10                47.90%                 40.10%                84.70%   
11                49.20%                 36.40%                88.60%   
12                   45%                 37.30%                86.80% 

In [4]:
hot_hand['Hot Hand Three-Point Percentage'] = hot_hand['Hot Hand Three-Point Percentage'].fillna('0%')
hot_hand['Team Record'] = hot_hand['Team Record'].replace(', ', '-', regex=True)
hot_hand['Hot Hand Team Record'] = hot_hand['Hot Hand Team Record'].replace(', ', '-', regex=True)
hot_hand

Players and Stats for 2023-24 Season  Points (per 100)  Rebounds (per 100)  \
0                 Giannis Antetokounmpo              41.6                15.7   
1                          Lebron James              34.6                 9.8   
2                           Bam Adebayo              28.2                15.2   
3                           Luka Doncic              43.3                11.8   
4                          Kevin Durant              35.5                 8.7   
5                     Tyrese Haliburton              29.5                 5.7   
6               Shai Gilgeous-Alexander              42.4                 7.8   
7                        Damian Lillard              33.1                 6.0   
8                          Nikola Jokic              37.8                17.7   
9                          Jayson Tatum              37.1                11.2   
10                        Jalen Brunson              40.9                 5.1   
11                         Devin Booker              36.7                 6.1   
12                         Tyrese Maxey              34.0                 4.8   
13                          Joel Embiid              50.9                16.2   
14                      Anthony Edwards              36.6                 7.7   
15                        Stephen Curry              39.1                 6.6   
16                     Donte Divincezno              26.9                 6.3   
17                         James Harden              23.8                 7.4   
18                        Kawhi Leonard              34.2                 8.8   
19                   Karl-Anthony Towns              32.9                12.6   

    Assists (per 100)  Blocks (per 100)  Steals (per 100)  \
0                 8.9               1.5               1.6   
1                11.2               0.7               1.7   
2                 5.7               1.4               1.7   
3                12.5               0.7               1.8   
4                 6.6               1.6               1.2   
5                16.0               1.0               1.8   
6                 8.8               1.3               2.8   
7                 9.5               0.3               1.3   
8                12.8               1.2               2.0   
9                 6.8               0.8               1.4   
10                9.6               0.2               1.3   
11                9.4               0.6               1.2   
12                8.1               0.6               1.3   
13                8.2               2.5               1.7   
14                7.2               0.7               1.8   
15                7.6               0.6               1.1   
16                4.6               0.8               2.3   
17               12.3               1.1               1.5   
18                5.2               1.3               2.4   
19                4.6               1.0               1.0   

   Field Goal Percentage Three-point Percentage Free Throw Percentage  \
0                 61.10%                 27.40%                65.70%   
1                 54.00%                 41.00%                75.00%   
2                 52.10%                 35.70%                75.50%   
3                 48.70%                 38.20%                78.60%   
4                 52.30%                 41.30%                85.60%   
5                 47.70%                 36.40%                85.50%   
6                 53.50%                 35.30%                87.40%   
7                 42.40%                 35.40%                   92%   
8                 58.30%                 35.90%                81.70%   
9                 47.10%                 37.60%                83.30%   
10                47.90%                 40.10%                84.70%   
11                49.20%                 36.40%                88.60%   
12                   45%                 37.30%                86.80% 

In [5]:
raw_player_data = pd.read_csv('Hot Hand Game Data(Sheet1).csv')
raw_player_data

Unnamed: 0  Points  Rebounds  Blocks  Steals Two Point Shots  \
0   Steph Curry      58         7     NaN     3.0           10;19   
1            KD      38         9     3.0     1.0           10;22   
2        LeBron      53        15     1.0     3.0           18;25   
3          Shai      73        16     NaN     4.0           27;42   
4         Jokic      55        31     3.0     2.0           21;30   
5           Ant      52         5     NaN     2.0           13;22   
6           KAT      62        17     1.0     3.0           14;25   
7       Giannis      68        25     2.0     3.0           27;39   
8          Book      44         3     NaN     NaN           12;22   
9           Bam      50        34     5.0     4.0           15;34   
10         Luka      72        10     NaN     2.0           16;19   
11   Haliburton      87         3     2.0     3.0           12;17   
12         Dame      71         4     1.0     2.0           10;14   
13        Tatum      37        14     1.0     1.0            8;15   
14      Brunson      47         7     NaN     1.0            6;15   
15        Maxey      39         2     1.0     2.0           11;19   
16       Embiid      73        17     4.0     2.0           22;36   
17   Divincenzo      32         7     NaN     3.0             2;4   
18       Harden      58        15     1.0     4.0           10;12   
19        Kawhi      25        11     3.0     3.0           10;17   

   Three Point Shots Free Throws  Points Scored (Team)  Points Allowed (Team)  \
0               9;18       13;14                   135                    135   
1                2;5       12;15                   121                    115   
2               4;10         5;7                   141                    156   
3                0;4       19;20                   194                    173   
4                3;8         4;8                   210                    212   
5               7;13         5;6                   143                    116   
6               7;14       13;13                   217                    207   
7                0;1       15;22                   162                    193   
8               5;14         5;6                   110                    116   
9                NaN       20;21                   197                    193   
10              9;22       15;19                   177                    167   
11             18;30         9;9                   194                    165   
12             13;25       18;20                   200                    164   
13              5;13         6;7                   149                    139   
14             10;18         5;7                   134                    162   
15              4;10         8;9                    94                    106   
16               2;4       23;24                   188                    163   
17              8;20         4;5                   186                    216   
18              7;20       17;18                   237                    244   
19              1;10         2;4                   135                    148   

    Total Possessions (Offense)  Total Possessions (Defense)  Team +/-  \
0                           113                          110       0.0   
1                            96                           96       6.0   
2                           131                          134       NaN   
3                           166                          166       NaN   
4                           195                          195       NaN   
5                           119                          121       NaN   
6                           174                          174       NaN   
7                           145                          145       NaN   
8                            94                           97       NaN   
9                           186                          189       NaN   
10                          128        

In [6]:
raw_player_data['Blocks'] = raw_player_data['Blocks'].fillna(0)
raw_player_data['Steals'] = raw_player_data['Steals'].fillna(0)
raw_player_data = raw_player_data.drop('Team +/-', axis=1)
raw_player_data['Three Point Shots'] = raw_player_data['Three Point Shots'].fillna('0;0')

In [7]:
raw_player_data = raw_player_data.rename(columns = {'Unnamed: 0': 'Player Name'})

In [8]:
raw_player_data['Two Point Shots'] = raw_player_data['Two Point Shots'].replace(';', '/', regex=True)
raw_player_data['Three Point Shots'] = raw_player_data['Three Point Shots'].replace(';', '/', regex=True)
raw_player_data['Free Throws'] = raw_player_data['Free Throws'].replace(';', '/', regex=True)

In [9]:
raw_player_data = raw_player_data.drop('Notes', axis=1)

In [10]:
raw_player_data

Player Name  Points  Rebounds  Blocks  Steals Two Point Shots  \
0   Steph Curry      58         7     0.0     3.0           10/19   
1            KD      38         9     3.0     1.0           10/22   
2        LeBron      53        15     1.0     3.0           18/25   
3          Shai      73        16     0.0     4.0           27/42   
4         Jokic      55        31     3.0     2.0           21/30   
5           Ant      52         5     0.0     2.0           13/22   
6           KAT      62        17     1.0     3.0           14/25   
7       Giannis      68        25     2.0     3.0           27/39   
8          Book      44         3     0.0     0.0           12/22   
9           Bam      50        34     5.0     4.0           15/34   
10         Luka      72        10     0.0     2.0           16/19   
11   Haliburton      87         3     2.0     3.0           12/17   
12         Dame      71         4     1.0     2.0           10/14   
13        Tatum      37        14     1.0     1.0            8/15   
14      Brunson      47         7     0.0     1.0            6/15   
15        Maxey      39         2     1.0     2.0           11/19   
16       Embiid      73        17     4.0     2.0           22/36   
17   Divincenzo      32         7     0.0     3.0             2/4   
18       Harden      58        15     1.0     4.0           10/12   
19        Kawhi      25        11     3.0     3.0           10/17   

   Three Point Shots Free Throws  Points Scored (Team)  Points Allowed (Team)  \
0               9/18       13/14                   135                    135   
1                2/5       12/15                   121                    115   
2               4/10         5/7                   141                    156   
3                0/4       19/20                   194                    173   
4                3/8         4/8                   210                    212   
5               7/13         5/6                   143                    116   
6               7/14       13/13                   217                    207   
7                0/1       15/22                   162                    193   
8               5/14         5/6                   110                    116   
9                0/0       20/21                   197                    193   
10              9/22       15/19                   177                    167   
11             18/30         9/9                   194                    165   
12             13/25       18/20                   200                    164   
13              5/13         6/7                   149                    139   
14             10/18         5/7                   134                    162   
15              4/10         8/9                    94                    106   
16               2/4       23/24                   188                    163   
17              8/20         4/5                   186                    216   
18              7/20       17/18                   237                    244   
19              1/10         2/4                   135                    148   

    Total Possessions (Offense)  Total Possessions (Defense)  Assists  \
0                           113                          110        5   
1                            96                           96        3   
2                           131                          134       11   
3                           166                          166       14   
4                           195                          195       31   
5                           119                          121        7   
6                           174                          174        9   
7                           145                          145       10   
8                            94                           97       10   
9                           186                          189       16   
10                          128                  

I had to clean the 'raw_player_data' dataset because I had put some missing values in that I wanted replaced with 0s; I also decided to get rid of the 'Team +/-' column because it was unimportant. I named the unnamed column 'player name', which I had forgotten to do previously and changed the semicolons to slashes to make the shooting splits look better. Finally, I removed the notes column so the dataset would remain solely numerical outside of player names.

In [11]:
pp100_sum = hot_hand['Points (per 100)'].sum()
pp100_sum

np.float64(719.1)

In [12]:
hhpp100_sum = hot_hand['Hot Hand Points (per 100)'].sum()
hhpp100_sum

np.float64(794.4999999999999)

In [13]:
(hhpp100_sum) - (pp100_sum)

np.float64(75.39999999999986)

In [14]:
((hhpp100_sum) - (pp100_sum)) / 20

np.float64(3.7699999999999934)

The 20 players combined scored 75.4 more points per 100 possessions when they had the hot hand; that averages out to a 3.77 points per 100 possessions increase per player.

In [15]:
ap100 = hot_hand['Assists (per 100)'].sum()
rp100 = hot_hand['Rebounds (per 100)'].sum()
bp100 = hot_hand['Blocks (per 100)'].sum()
sp100 = hot_hand['Steals (per 100)'].sum()

In [16]:
print (ap100)
print (rp100)
print (bp100)
print (sp100)

175.6
191.20000000000002
19.900000000000002
32.9


In [17]:
hhap100 = hot_hand['Hot Hand Assists (per 100)'].sum()
hhrp100 = hot_hand['Hot Hand Rebounds (per 100)'].sum()
hhbp100 = hot_hand['Hot Hand Blocks (per 100)'].sum()
hhsp100 = hot_hand['Hot Hand Steals (per 100)'].sum()

In [18]:
print (hhap100)
print (hhrp100)
print (hhbp100)
print (hhsp100)

163.69999999999996
169.60000000000002
19.4
33.599999999999994


For the other 4 main stats - rebounds, assists, blocks, and steals - steals is the only one that increases with the hot hand.